platforms/core-configuration/input-tracking/src/main/java/org/gradle/internal/configuration/inputs/AccessTrackingProperties.java

```diff
@@ -185,7 +185,7 @@ public void replaceAll(BiFunction<? super Object, ? super Object, ?> function) {
     }
 
     @Override
-    public Object putIfAbsent(Object key, Object value) {
+    public synchronized Object putIfAbsent(Object key, Object value) {
         Object oldValue;
         synchronized (delegate) {
```

Comment:
🤔 Since `delegate` is also a `Properties` instance (thus `synchronized`) and we're now synchronizing this `Properties` instance, what's the point of `synchronized (delegate)` here?

Reply:
> 🤔 Since `delegate` is also a `Properties` instance (thus `synchronized`) and we're now synchronizing this `Properties` instance, what's the point of `synchronized (delegate)` here?

Well, making these methods synchronized doesn't change much in this regard. I guess, I was either over-defensive when writing these methods and didn't look if the method is synchronized or not, or this is a leftover of some earlier code state when there were some extra steps in the synchronized block.

TBH, the synchronization contract is still broken for system properties, because user code gets a new Properties instance every time. I'd rather replace this fix with a class-level suppression and address synchronizations concerns separately.